central/networkgraph/service/service_impl_postgres_test.go

```diff
@@ -0,0 +1,509 @@
+		testutils.GetExtSrcNetworkEntity(entityIDs[2], "ext3", "10.0.100.25/32", false, testCluster),
+	}
+
+	deployments := []*storage.Deployment{
+		&storage.Deployment{
+			Id:        fixtureconsts.Deployment1,
+			ClusterId: testCluster,
+			Namespace: testNamespace,
+		},
+
+		&storage.Deployment{
+			Id:        fixtureconsts.Deployment2,
+			ClusterId: testCluster,
+			Namespace: testNamespace,
+		},
+	}
+
+	for _, deployment := range deployments {
+		err := s.deploymentsDataStore.UpsertDeployment(globalWriteAccessCtx, deployment)
+		s.NoError(err)
+	}
+
+	for _, e := range entities {
+		err := s.entityDataStore.CreateExternalNetworkEntity(globalWriteAccessCtx, e, true)
+		s.NoError(err)
+	}
+
+	singleEntityFlow := externalFlow(deployments[0], entities[0], false)
+	multiEntityFlows := []*storage.NetworkFlow{
+		externalFlow(deployments[0], entities[2], false),
+		externalFlow(deployments[1], entities[2], false),
+	}
+
+	allFlows := []*storage.NetworkFlow{singleEntityFlow}
+	allFlows = append(allFlows, multiEntityFlows...)
+
+	flowStore, err := s.flowDataStore.CreateFlowStore(globalWriteAccessCtx, testCluster)
+	s.NoError(err)
+
+	err = flowStore.UpsertFlows(globalWriteAccessCtx, allFlows, timestamp.FromGoTime(time.Now()))
+	s.NoError(err)
+
+	for _, tc := range []struct {
+		name          string
+		request       *v1.GetExternalNetworkFlowsRequest
+		expected      *v1.GetExternalNetworkFlowsResponse
+		expectSuccess bool
+	}{
+		{
+			name: "Get single entity flows",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  entityIDs[0],
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected: &v1.GetExternalNetworkFlowsResponse{
+				Entity: entities[0].GetInfo(),
+				Flows: []*storage.NetworkFlow{
+					singleEntityFlow,
+				},
+			},
+			expectSuccess: true,
+		},
+		{
+			name: "Entity with no flows",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  entityIDs[1],
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected: &v1.GetExternalNetworkFlowsResponse{
+				Entity: entities[1].GetInfo(),
+				Flows:  []*storage.NetworkFlow{},
+			},
+			expectSuccess: true,
+		},
+		{
+			name: "Invalid entity ID",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  "invalid ID",
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected:      nil,
+			expectSuccess: false,
+		},
+		{
+			name: "Invalid cluster",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: "invalid cluster",
+				EntityId:  entityIDs[0],
+				Query:     fmt.Sprintf("Namespace:%s", testNamespace),
+			},
+			expected:      nil,
+			expectSuccess: false,
+		},
+		{
+			name: "entity with multiple flows",
+			request: &v1.GetExternalNetworkFlowsRequest{
+				ClusterId: testCluster,
+				EntityId:  entityID3.String(),
```

Comment:
Just for consistency
```suggestion
				EntityId:  entityIDs[2].String(),
```